libs/offline-tasks/src/main/java/org/opensearch/offline_tasks/TaskClient.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.offline_tasks;
+
+import org.opensearch.common.annotation.ExperimentalApi;
+import org.opensearch.offline_tasks.task.Task;
+import org.opensearch.offline_tasks.task.TaskId;
+
+import java.util.List;
+
+/**
+ * Client used to interact with Task Store/Queue
+ *
+ * @opensearch.experimental
+ */
+@ExperimentalApi
+public interface TaskClient {
+
+    /**
+     * Submit a new task to TaskStore/Queue
+     *
+     * @param task Task to be submitted for execution on offline nodes
+     */
+    void submitTask(Task task);
```

Comment:
Should we return a boolean to indicate success/failure of the operation?
```suggestion
    boolean submitTask(Task task);
```

Reply:
i don't think boolean should be needed. No exception while method call should indicate a successful submission. Although i do think we might want to add support for async call to these methods. let me think on this